vulture_os/system/cluster/api.py

```diff
@@ -170,6 +171,120 @@ def secret_key(request):
         }, 500)
 
 
+@api_need_key('cluster_api_key')
+@require_http_methods(['GET'])
+def get_message_queues(request):
+    try:
+        params = {}
+        status = request.GET.get('status')
+        if status:
+            if status not in ("new", "running", "done", "failure"):
+                return JsonResponse({
+                    'status': False,
+                    'data': _("Invalid parameter 'status'")
+                }, status=400)
+            params['status'] = status
+        node = request.GET.get('node')
+        if node:
+            params['node__name'] = node
+
+        tasks = MessageQueue.objects.filter(**params).order_by('modified')
+        limit = request.GET.get('limit', "100")
+        try:
+            limit = int(limit)
+        except:
+            return JsonResponse({
+                'status': False,
+                'data': _("Invalid parameter 'limit'")
+            }, status=400)
+        tasks = tasks[:limit]
+
+        res = [m.to_template() for m in tasks]
+
+        return JsonResponse({
+            'status': True,
+            'data': res
+        })
+
+    except Exception as e:
+        logger.critical(e, exc_info=1)
+        if settings.DEV_MODE:
+            raise
+
+        return JsonResponse({
+            'status': False,
+            'data': _('An error has occurred')
+        }, 500)
+
+
+@api_need_key('cluster_api_key')
+@require_http_methods(['GET'])
+def get_cluster_status(request):
+    try:
+        params = {}
+        node_name = request.GET.get('node')
+        node = None
+        if node_name:
+            try:
+                node = Node.objects.get(name=node_name)
+                params['node'] = node
+            except Node.DoesNotExist:
+                return JsonResponse({
+                    'status': False,
+                    'data': _("Invalid parameter 'node'")},
+                    status=404
+                )
+
+        pending_tasks = MessageQueue.objects.filter(**params, status__in=["new", "running"]).count()
+        failed_tasks = MessageQueue.objects.filter(**params, status="failure").count()
+
+        result = True
+        frontends_status = []
+        for f in Frontend.objects.filter(enabled=True):
+            if node:
+                if node not in f.get_nodes():
+                    continue
+                f_status = f.status.get(node.name, "UNKNOWN")
+                if f_status != "OPEN":
+                    logger.info(f_status)
```

Comment:
logger seems useless as 'info' without a bit of context around

Reply:
Logger removed. Thks.